unit-tests/dds/test-librs-options.py

```diff
@@ -93,8 +100,57 @@
         s.set_option_value( eo, 'Last' )
         test.check_equal( s.get_option( eo ), 1. )
 
+    with test.closure( 'Play with read-only option' ):
+        ro = next( o for o in options if str(o) == 'R/O Option' )
+        test.check_throws( lambda:
+            s.get_option( ro ),
+            RuntimeError, 'use rs2_get_option_value to get string values' )
+        rv = s.get_option_value( ro )
+        test.check_equal( rv.type, rs.option_type.string )
+        test.check_equal( rv.value, 'Value' )
+        test.check( s.is_option_read_only( rv.id ) )
+        rr = s.get_option_range( rv.id )
+        test.check_equal( rr.min, 0. )
+        test.check_equal( rr.max, 0. )
+        test.check_equal( rr.default, 0. )
+        test.check_equal( rr.step, 0. )
+        test.check_throws( lambda:
+            s.set_option( ro, 2. ),
+            RuntimeError, 'use rs2_set_option_value to set string values' )
+        test.check_throws( lambda:
+            s.set_option_value( ro, 'Blah' ),
+            RuntimeError, 'option is read-only: R/O Option' )
+
+    with test.closure( 'Check serialization through serialized_device' ):
+        sdev = rs.serializable_device( dev )
+        test.check( sdev, on_fail=test.RAISE )
+        import json
+        j = json.loads( sdev.serialize_json() )
+        test.info( "serialize_json()", j )
+        params = j.get( 'parameters', [] )
+        test.check_equal( len( params ), 5 )  # Without FRAMES_QUEUE_SIZE
+        test.check_equal( params.get( 'sensor/Boolean Option' ), True )
+
+        # Confirm previous value was set
+        bv = s.get_option_value( bo )
+        test.check_equal( bv.type, rs.option_type.boolean )
+        test.check_equal( bv.value, True )
+        test.check_equal( s.get_option( bo ), 1. )
+
+        # Change using load_json() and verify
+        params.update({ 'sensor/Boolean Option': False })
+        log.d( f'updated json: {j}' )
+        remote.capture_stdout()
+        sdev.load_json( json.dumps( j ) )
+        rv = remote.get_stdout()
+        test.check_equal( s.get_option( bo ), 0. )
+        test.info( "option change order", rv )
+        test.check_equal( len(rv), 5 )              # all options should have been changed
+        test.check_equal( rv[0], 'Visual Preset' )  # but Visual Preset should always update first
+        del sdev
+
     with test.closure( 'All done' ):
-        dev = None
-        context = None
+        del dev
```

Comment:
Is calling `del` better then setting `None`? We have this in other places too

Reply:
There is no difference for garbage collection; but after `del` then the variable no longer exists. I find it clearer.